core/iwasm/libraries/lib-pthread/lib_pthread_wrapper.c

```diff
@@ -160,6 +178,12 @@ thread_info_destroy(void *node)
             os_cond_destroy(info_node->u.cond);
         wasm_runtime_free(info_node->u.cond);
     }
+#if WASM_ENABLE_LIB_PTHREAD_SEMAPHORE != 0
+    else if (info_node->type == T_SEM) {
+        if (info_node->status != SEM_DESTROYED)
+            os_sem_close(info_node->u.sem);
```

Comment:
Should also `os_sem_unlink(info_node->u.sem)`?

Reply:
The semaphore may be linked out side the wasm app, but it's a problem if user forget to unlink the sem.

Reply:
After check the spec by https://man7.org/linux/man-pages/man3/sem_unlink.3.html, it's seems safe to unlink sem here since semaphore will be destroied on all sem closed.

Reply:
OK, so no need to modify it here, right?

Reply:
Yes, I thinks so